pkg/apis/mayadata.io/v1alpha1/backup_schedule.go

```diff
@@ -0,0 +1,212 @@
+type DMaaSBackupStatus struct {
+	// Phase represents the current phase of DMaaSBackup
+	Phase DMaaSBackupPhase `json:"phase"`
+
+	// Reason represents the cause of failure in DMaaSBackup
+	Reason string `json:"reason,omitempty"`
+
+	// Message represents the cause/action/outcome of DMaaSBackup
+	Message string `json:"message,omitempty"`
+
+	// VeleroSchedules represents the list of Velero Schedule created by DMaaSBackup
+	// +nullable
+	VeleroSchedules []VeleroScheduleDetails `json:"veleroSchedules,omitempty"`
+
+	// VeleroBackupName represents the name of Velero Backup, created by DMaaSBackup,
+	// if VeleroScheduleSpec is having empty schedule
+	VeleroBackupName string `json:"veleroBackupName,omitempty"`
+
+	// LatestBackupStatus represents the status of latest backup created by DMaaSBackup
+	LatestBackupStatus LatestBackupStatusDetails `json:"latestBackukpStatus,omitempty"`
+}
+
+// VeleroScheduleStatus represents the status of VeleroSchedule
+type VeleroScheduleStatus string
+
+const (
+	// Active represents the active state of VeleroSchedule
+	Active VeleroScheduleStatus = "Active"
+
+	// Deleted represents the deleted VeleroSchedule
+	Deleted VeleroScheduleStatus = "Deleted"
+
+	// Erased represents the VeleroSchedule,for which data at remote storage is deleted
+	Erased VeleroScheduleStatus = "Erased"
+)
+
+// VeleroScheduleDetails represents the information about schedule
+type VeleroScheduleDetails struct {
+	// ScheduleName represents the velero schedule resource name
+	ScheduleName string `json:"scheduleName"`
+
+	// CreationTimestamp defines the time-stamp of velero schedule CR
+	// +nullable
+	CreationTimestamp *metav1.Time `json:"creationTimestamp,omitempty"`
+
+	// Status represents the velero schedule status
+	Status VeleroScheduleStatus `json:"status"`
+}
+
+// LatestBackupStatusDetails represents the status of Latest backup
+type LatestBackupStatusDetails struct {
+	// BackupName represents the name of the backup, which is in-progress or completed recently
+	BackupName string `json:"backupName"`
+
+	// Phase represents the given Backup Phase
+	Phase string `json:"phase"`
+
+	// StartTimestamp represents the Backup start time
+	// +nullable
+	StartTimestamp *metav1.Time `json:"startTimestamp,omitempty"`
+
+	// CompletionTimestamp represents the completion time of the given Backup
+	// +nullable
+	CompletionTimestamp *metav1.Time `json:"completionTimestamp,omitempty"`
+
+	// SnapshotStatus represents the list of snapshotStatus
+	// +nullable
+	SnapshotStatus []SnapshotStatusDetails `json:"snapshotStatus,omitempty"`
+}
+
+// SnapshotType defines the type of snapshot
+type SnapshotType string
+
+const (
+	// SnapshotTypeRestic represents the restic base snapshot
+	SnapshotTypeRestic SnapshotType = "Restic"
+
+	// SnapshotTypeCStor represents the cstor base snapshot
+	SnapshotTypeCStor SnapshotType = "cStor"
+)
+
+// SnapshotStatusDetails represents the snapshot information and it's status
+type SnapshotStatusDetails struct {
```

Comment:
lets add pvName also (can be next PR)

Reply:
similar can be the case for PVRestoreStatusDetails also..
we can add pvName or pvcName if available
However, we can take in next PR